inc/auth.class.php

```diff
@@ -846,6 +846,14 @@ function login($login_name, $login_password, $noauto = false, $remember_me = fal
                      || $this->user->fields["authtype"] == $this::LDAP) {
 
                   if (Toolbox::canUseLdap()) {
+                     // Attempted to retrieve auths_id via login_name.
+                     // This allows the user to be authenticated against their LDAP directory
+                     // rather than testing all active LDAP directories in GLPI.
+                     if ($this->user->fields["auths_id"] == 0) {
+                        if ($this->user->getFromDBbyName(addslashes($login_name))) {
+                           $this->user->fields["auths_id"] = $this->user->getField('auths_id');
+                        }
+                     }
```

Comment:
What would happen in a case of a LDAP migration ?

Unless I am wrong, with the actual logic, the user `auths_id` will be automatically updated when this user has been moved from a LDAP server A to a LDAP server b. With this new logic, will this still work ?